packages/react/src/analytics/integrations/shared/GoogleConsentMode/__tests__/GoogleConsentMode.test.ts

```diff
@@ -0,0 +1,50 @@
+import { GoogleConsentMode, type GoogleConsentModeMappings } from '../index.js';
+import type { ConsentData } from '@farfetch/blackout-analytics';
+
+describe('GoogleConsentMode', () => {
+  const dataLayerName = 'dataLayer';
+  const mockConsent: ConsentData = {
+    consent_a: true,
+    consent_b: true,
+    consent_c: true,
+    consent_d: true,
+  };
+  let consentMappings: GoogleConsentModeMappings;
+
+  beforeEach(() => {
+    window[dataLayerName] = [];
+
+    consentMappings = {
+      ad_storage: ['consent_a', 'consent_b'],
+      ad_user_data: ['consent_a'],
+      ad_personalization: ['consent_c'],
+      analytics_storage: [],
+    };
+  });
+
+  it('should initialize GoogleConsentMode with all mappings granted', () => {
+    const googleConsent = new GoogleConsentMode(
+      dataLayerName,
+      mockConsent,
+      consentMappings,
+    );
+
+    expect(googleConsent).toBeInstanceOf(GoogleConsentMode);
+
+    expect(window.dataLayer).toMatchSnapshot();
+  });
+
+  it('should initialize GoogleConsentMode without consents and grant consents before', () => {
```

Comment:
Is this description correct? It seems it is missing something.

Reply:
Updated description to: Initialize GoogleConsentMode without any consents. Later, set all consents and expect that all results are granted.